File: csharp/matching-brackets/MatchingBrackets.cs
using System.Collections.Generic;
using System.Text.RegularExpressions;

public static class MatchingBrackets
{
    private static readonly Dictionary<char, char> _bracketPairs = new()
    {
        [']'] = '[',
        [')'] = '(',
        ['}'] = '{'
    };

    public static bool IsPaired(string input)
    {
        input = new Regex(@"(?![\[\]{}()]).").Replace(input, "");
        var leftBrackets = new Stack<char>(" ");

        foreach (var ch in input)
        {
            if (_bracketPairs.GetValueOrDefault(ch, '*') == leftBrackets.Peek())
                leftBrackets.Pop();
            else
                leftBrackets.Push(ch);
        }

        return leftBrackets.Count == 1;
    }
}

File: solutions/julia/matching-brackets/1/matching-brackets.jl
matches = Dict('}' => '{', ')' => '(', ']' => '[')

function matching_brackets(str)
    str = replace(str, r"(?![\[\]\{\}\(\)])." => "")

    bs = [' ']
    for c in collect(str)
        bs[end] == get(matches, c, nothing) ? pop!(bs) : push!(bs, c)
    end

	length(bs) == 1
end